utils/src/utils/AzExtFsExtra.ts

```diff
@@ -81,14 +81,14 @@ export namespace AzExtFsExtra {
         return JSON.parse(file) as T;
     }
 
-    export async function writeJSON(resource: Uri | string, contents: string | unknown): Promise<void> {
+    export async function writeJSON(resource: Uri | string, contents: string | unknown, spaces?: string | number): Promise<void> {
```

Comment:
Should we define a default value (e.g. 2 or 4)? If the purpose is for the JSON to be written to a file it seems like it would almost always (or at least, usually) be desirable to have whitespace be used. I think that _not_ wanting whitespace would be the exception, and for that we could require the caller to pass in a [zero value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#parameters):
> Values less than 1 indicate that no space should be used.

Reply:
Yeah, that makes sense to me.